packages/plugins/EVM/src/state/TransactionFormatter/utils.ts

```diff
@@ -1,14 +1,46 @@
 import { FungibleToken, scale10, formatBalance } from '@masknet/web3-shared-base'
 import type { ChainId, SchemaType } from '@masknet/web3-shared-evm'
+import { decodeEvents, createContract } from '@masknet/web3-shared-evm'
+import { Web3StateSettings } from '../../settings'
+import BigNumber from 'bignumber.js'
+import type { AbiItem } from 'web3-utils'
 
-export function getTokenAmountDescription(
-    amount = '0',
-    token?: FungibleToken<ChainId, SchemaType>,
-    negative?: boolean,
-) {
-    const symbol = negative ? '- ' : ''
+export function getTokenAmountDescription(amount = '0', token?: FungibleToken<ChainId, SchemaType>) {
     const value = scale10(1, 9 + (token?.decimals ?? 18)).isGreaterThanOrEqualTo(amount)
         ? formatBalance(amount, token?.decimals ?? 0, 4)
         : 'infinite'
-    return `${symbol}${value} ${token?.symbol?.trim()}`
+
+    if (value !== 'infinite' && new BigNumber(value).isLessThan(new BigNumber('0.000001'))) {
+        return `<0.000001 ${token?.symbol?.trim()}`
+    }
+    return `${value} ${token?.symbol?.trim()}`
+}
+
+export class DescriptorWithTransactionReceipt {
```

Comment:
Get `receipt` while creating the context.

Reply:
And more, It is about to get the decoded results from receipt by abi in `DescriptorWithTransactionReceipt`.